### ISSUE TITLE
Native Windows Support

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#1237]
## Description

Support compiling Lighthouse with MSVC and executing as a native binary on Windows, without requiring the Windows Subsystem for Linux. Windows support under WSL is covered by #1238 

## Known Issues

- [x] ~~LevelDB doesn't run on Windows. Soln: RocksDB, LMDB? (#483)~~
- [ ] We use some `unix` filesystem operations, particularly around permissions. Soln: separate permissions code for Windows.
- [ ] Probably lots of networking incompatibilities (to investigate)
- [x] `milagro_bls` uses paths that are not valid on Windows (#1447)

More points to be added as they're discovered

**Comment by michaelsproul** [comment]
As a temporary workaround for the LevelDB issue, we could use the MemoryStore on Windows, and conditionally compile out all references to LevelDB using a feature flag (https://doc.rust-lang.org/cargo/reference/features.html).

**Comment by paulhauner** [comment]
I believe @AgeManning is having a look at this, when he has spare time.

**Comment by AgeManning** [comment]
Yep, conditional on the spare time part. Things keep arising at the moment. 

If someone is interested in this, feel free to get a start on this.

**Comment by pawanjay176** [comment]
Managed to make everything compile (without restricting file permissions) and run a node and validator.

Working on fixing the file permissions atm.

**Comment by pawanjay176** [comment]
Found a nice library for manipulating windows file access control lists https://github.com/trailofbits/windows-acl

**Comment by remyroy** [comment]
My naive attempt at compiling Lighthouse 1.2.2 gave me this output on Windows 10.

```
error[E0308]: mismatched types
   --> validator_client\slashing_protection\src\slashing_database.rs:137:48
    |
137 |     fn set_db_file_permissions(file: &File) -> Result<(), NotSafe> {}
    |        -----------------------                 ^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found `()`
    |        |
    |        implicitly returns `()` as its body has no tail or `return` expression
    |
    = note:   expected enum `std::result::Result<(), NotSafe>`
            found unit type `()`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `slashing_protection`
```

WSL is very far from ideal to run Lighthouse on Windows. Full support and official binary for Windows would be appreciated.

**Comment by michaelsproul** [comment]
Closed by #2333, thank you @ethDreamer! :tada:
